### ISSUE TITLE
Postgres 9.5 native upsert support?

### ISSUE THREAD

**Issue by nightsurge** [seamusabshere/upsert#57]
Postgres 9.5 added Native Upsert support:

http://www.craigkerstiens.com/2015/05/08/upsert-lands-in-postgres-9.5/

```
INSERT INTO products (
    upc, 
    title, 
    description, 
    link) 
VALUES (
    123456789, 
    ‘Figment #1 of 5’, 
    ‘THE NEXT DISNEY ADVENTURE IS HERE - STARRING ONE OF DISNEY'S MOST POPULAR CHARACTERS! ’, 
    ‘http://www.amazon.com/dp/B00KGJVRNE?tag=mypred-20’
    )
ON CONFLICT DO UPDATE SET description=excluded.description;
```

**Comment by seamusabshere** [comment]
@nightsurge we should detect it and use if available... maybe.

as i understand it, it requires the presence of a unique index - which this library does not.

so, it's not a drop-in replacement.

i would be happy to hear suggestions!

**Comment by rocketraceheroes** [comment]
I will update once it is out of beta and give it a try!

**Comment by amalagaura** [comment]
@seamusabshere Your approach is nicer to not require the unique index, but in real world applications I can see having the unique index being an acceptable pre-condition.

**Comment by pnomolos** [comment]
@nightsurge @amalagaura See #78.  It's in progress now :)

**Comment by pnomolos** [comment]
Closed via #79.  Will be used if it's available (dependent on PG version as well as existence of a unique index on the column(s) being used for selectors).
